src/playdate/nineSlice.nim

```diff
@@ -0,0 +1,171 @@
+import graphics
+import std/[importutils, bitops, math, strutils]
+
+privateAccess(BitmapData)
+privateAccess(LCDBitmap)
+
+type
+    NineSliceRow {.byref.} = object
+        ## The bytes for a single row within a nine slice
+        leftBytes: seq[uint8]
+        middleBytes: seq[uint8]
+        rightBytes: seq[uint8]
+        leftRightBitLen: int
+
+    NineSliceData {.byref.} = object
+        ## Stores the rows for a single nine slice bitmap
+        top: seq[NineSliceRow]
+        middle: seq[NineSliceRow]
+        bottom: seq[NineSliceRow]
+
+    NineSlice* = ref object
+        ## A precalculated nine slice
+        image: NineSliceData
+        case hasMask: bool
```

Comment:
We should maybe consider the approach with a nullable `mask` field and a `hasMask` getter proc?

Reply:
Sounds good. I’ll update.